src/library/export/trackexportdlg.cpp

```diff
@@ -48,6 +52,47 @@ void TrackExportDlg::showEvent(QShowEvent* event) {
         qDebug() << "Programming error: did not initialize m_exporter, about to crash";
         return;
     }
+
+    //Checks if there are any files that are missing. If so, it will throw dialog informing the user.
+    //User can then decide to either cancel the export or skip those files
+    QList<QString> files = m_worker->getMissingTracks();
+    {
+        if (!files.isEmpty()) {
+            QDialog notExportedDlg = QDialog();
+            notExportedDlg.setWindowTitle(tr("Export Track Files"));
+            QVBoxLayout notExportedLayout;
```

Comment:
The layout elements have to be allocated dynamically because QVBoxLayout takes ownership and deletes then on destruction.

https://doc.qt.io/qt-5/qlayout.html#addItem

Reply:
so if i'm understanding this correctly, only the dialog should be allocated in stack, and since everything else goes into the QVBoxLayout, they should be allocated in heap?

Reply:
Yes. Only the top-level object could be placed on the stack. Passing stack pointers of other members in the object hierarchy to Qt might cause undefined behavior depending on how Qt handles them. Ownership can only be passed for dynamically allocated memory.